tests/simple/simple_test.go

```diff
@@ -113,30 +130,95 @@ func TestMain(m *testing.M) {
 	os.Exit(m.Run())
 }
 
-func TestSimpleFile(t *testing.T) {
-	// Special case to handle test invocation without args. See TestMain()
-	// early return comment.
-	if rc == nil {
-		return
-	}
-	// Run the flag-configured tests.
-	for _, filename := range flag.Args() {
-		testFile, err := parseSimpleFile(filename)
-		if err != nil {
-			t.Fatalf("Can't parse input file %v: %v", filename, err)
+ func TestSimpleFile(t *testing.T) {
+        // Special case to handle test invocation without args. See TestMain()
+        // early return comment.
+        if rc == nil {
+                return
+        }
+	var skipTests []string
+	firstFlag := true
+	for _, flagVal := range skipFlags {
+		if !strings.Contains(flagVal, "/") {
+			log.Fatal("Unable to parse skip_test flag, specifically for ", flagVal)
+		}
+		fileInd := strings.Index(flagVal, "/")
+		if (flagVal[:fileInd] == ""){
+			log.Fatal("Empty filename")
 		}
-		t.Logf("Running tests in file %v\n", testFile.Name)
-		for _, section := range testFile.Section {
-			t.Logf("Running tests in section %v\n", section.Name)
-			for _, test := range section.Test {
-				desc := fmt.Sprintf("%s/%s/%s", testFile.Name, section.Name, test.Name)
-				t.Run(desc, func(t *testing.T) {
-					err := rc.RunTest(test)
-					if err != nil {
-						t.Fatal(err)
+		allSections := strings.Split(flagVal, ";")
+		filename := flagVal[:fileInd+1]
+		for _, sectionVal := range allSections {
+			if firstFlag {
```

Comment:
I don't think you need a "firstFlag".  Consider the following approach:

Starting with the flag value `f1/s1/t1,t2;s2`, use `strings.SplitN(flagVal, "/", 2)` to split into `filename = 'f1'` and `s = 's1/t1,t2;s2'`.  It's an error if `filename` or `s` is empty.

Now split `s` by `';'` and iterate on each section.

For each section, see if it has a `'/'`.  If not then the section is just a section name.  If it does have a `'/'`, then the left part is the section name, and the right part can be split on `','` to get the list of tests.

Reply:
Would you ever have a flag like f1/s1/t1,t2;s2/t1?

Reply:
Sure.  So you'd:

1.  Split on the first "/" to get the filename `f1` and the list of section parts.
2.  Split the section parts on ";" to get the individual sections.
3.  For each section, split on the first "/" to get the section name and the list of test parts (or if no "/", then it's a raw section name.
4.  Split the test parts on "," to get the test names.